packages/components/src/unit-control/index.tsx

```diff
@@ -100,7 +100,7 @@ function UnforwardedUnitControl(
 		if ( parsedUnit !== undefined ) {
 			setUnit( parsedUnit );
 		}
-	}, [ parsedUnit ] );
+	}, [ parsedUnit, setUnit ] );
```

Comment:
Yup!

And there's also [`useControlledValue`](https://github.com/WordPress/gutenberg/blob/5cca2e1a8021679aaf2fc713c1a2c1cce168c374/packages/components/src/utils/hooks/use-controlled-value.ts) 🤦  It would be good to see if we can use only one implementation across the whole package